### PR TITLE
Fix/issue 186

### LINKED ISSUE #186: confusing burn-in setting in "Discrete: rates "
For the user it may not be immediately clear that the burn-in needs to be set as a fraction, so please label it as 'select burn-in fraction'. Also, there is no scale shown for this fraction (under the tick marks) and the highlighted value is confusing. For example, at times, it shows the value of '0.1' at the first tick mark, which would make sense:
burn-in_0 1
However, after fiddling with it, the value at the first tick mark became '0.0014'?:  
burn-in_0 0014
checking the completed analysis indicates it was actually 0.004 something?
burn-in

### PR DESCRIPTION
**Pull request by fbielejec**
closes #186

### CODE REVIEW

File: src/clj/api/resolvers.clj [hunk 1]
```diff
@@ -70,7 +70,9 @@
 (defn get-bayes-factor-analysis
   [{:keys [db]} {id :id :as args} _]
   (log/info "get-bayes-factor-analysis query" args)
-  (clj->gql (bayes-factor-model/get-bayes-factor-analysis db {:id id})))
+  (let [response (bayes-factor-model/get-bayes-factor-analysis db {:id id})]
+    (log/info "get-bayes-factor-analysis response" response)
```
Comment (jpmonettas) [inline src/clj/api/resolvers.clj]: shouldn't this be log/debug?

File: src/cljs/ui/events/graphql.cljs [hunk 2]
```diff
@@ -236,11 +236,15 @@
   {:db (update-in db [:analysis id] merge analysis)})
 
 (defmethod handler :get-bayes-factor-analysis
-  [{:keys [db]} _ {:keys [id] :as analysis}]
-  {:db
-   (-> db
-       (update-in [:analysis id] merge (:analysis analysis))
-       (update-in [:analysis id] merge analysis))})
+  [{:keys [db]} _ {:keys [id burn-in] :as analysis}]
+    (prn "@handler 1" burn-in)
```
Comment (jpmonettas) [inline src/cljs/ui/events/graphql.cljs]: I think this can be removed

File: src/cljs/ui/events/graphql.cljs [hunk 3]
```diff
@@ -236,11 +236,15 @@
...
+  (let [;; fix for weird JS behaviour, where it will parse floats with full precision
+        burn-in (round burn-in 2)]
+    (prn "@handler 2" burn-in)
```
Comment (jpmonettas) [inline src/cljs/ui/events/graphql.cljs]: I think this can be removed

File: src/cljs/ui/new_analysis/discrete_rates.cljs [hunk 4]
```diff
@@ -44,49 +43,53 @@
              :or   {burn-in 0.1}}
             @bayes-factor
             controls-disabled? (or (seq? @field-errors)
-                                   (not log-file-name))]
+                                   (not log-file-name))
+            ;; weird JS behaviour, where it will parse floats with full precision
+            burn-in (round burn-in 2)
```
Comment (jpmonettas) [inline src/cljs/ui/new_analysis/discrete_rates.cljs]: I don't get the precision removing thing? can't it just have all the precision and do a format for displaying on the slider?

Comment (fbielejec) [reply src/cljs/ui/new_analysis/discrete_rates.cljs]: It makes a round-trip when setting is sent to the server - mutations response is handled by the graphql event handler and put into app db, such that when starting the analysis the db value could differ from the displayed - confusing for the user.

Comment (jpmonettas) [reply src/cljs/ui/new_analysis/discrete_rates.cljs]: but why do we need to remove precision? that is what I don't understand

File: src/clj/api/resolvers.clj (hunk 1 above)
Comment (fbielejec) [reply src/clj/api/resolvers.clj]: fixed

File: src/cljs/ui/events/graphql.cljs (hunk 2 above)
Comment (fbielejec) [reply src/cljs/ui/events/graphql.cljs]: done

File: src/cljs/ui/events/graphql.cljs (hunk 3 above)
Comment (fbielejec) [reply src/cljs/ui/events/graphql.cljs]: done

File: src/cljs/ui/new_analysis/discrete_rates.cljs (hunk 4 above)
Comment (fbielejec) [reply src/cljs/ui/new_analysis/discrete_rates.cljs]: The slider increments by 0.1, all the extra digits are just some artifacts of JS handling numbers (e.g. 0.2 returned in a server response becomes 0.2000000000000000019)

File: src/cljs/ui/new_analysis/discrete_rates.cljs [hunk 5]
```diff
@@ -95,20 +96,21 @@
...
-               [slider {:value             burn-in
-                        :min               0.0
-                        :max               0.9
-                        :step              0.1
-                        :valueLabelDisplay :auto
-                        :marks             true
-                        :on-change         (fn [_ value]
-                                             (debounce (>evt [:bayes-factor/set-burn-in value]) 10))}]]]])]
+               [mui-slider {:step      0.1
+                            :class     "speed-slider"
```
Comment (fbielejec) [inline src/cljs/ui/new_analysis/discrete_rates.cljs]: remove

**Review by jpmonettas — APPROVED** [review]

**Pull request merged**